### PR TITLE
Add rets status and correct max rows

### PR DESCRIPTION
**Pull request by kaloramik**
Rets has max row limitations, and will sometimes limit the number of results that it returns. The usual symptom here is that count > len(data)

It will indicate when this is out of sync in 2 ways 

- existence of maxrows tag
- rets-status 

This PR fixes the maxrows check and parses out rets-status when it exists

### CODE REVIEW

**Comment by kaloramik** [comment]
Also made lookup multi return a list, as tuple doesnt work when sqlalchemy is trying to insert

**Comment by kaloramik** [comment]
removed status parsing for now, but eventually I think we should expose both rets code / replytext somehow. A new user to the rets client would never even know this behavior exists, and they would only find out by seeing that count != len(data) 

Also, the RETS-STATUS actually SHOULD be different than the top level rets status 

The rets-status MAY be included in the response if the ReplyCode or ReplyText given
in the body-start-line becomes invalid during the creation of the response. If the server
includes a rets-status in its reply, the client MUST use the ReplyCode and ReplyText
from the rets-status rather than from the body-start-line.


This is true in my case, the response looks like 

<RETS ReplyCode="0" ReplyText="SUCCESS" >\r\n<COUNT Records="6673"/>\r\n<DELIMITER value="09" />\r\n<COLUMNS>
...
</DATA>\r\n<MAXROWS/>\r\n<RETS-STATUS ReplyCode="20208" ReplyText="Maximum Records Exceeded." />\r\n</RETS>

**Comment by kaloramik** [comment]
actually, we should just raise when theres rets-status I think, no one really wants to use the result if theres a status...

File: rets/http/parsers/parse.py [hunk 1]
```diff
@@ -124,7 +124,8 @@ def parse_search(response: Response) -> SearchResult:
 
     return SearchResult(
         count=count,
-        max_rows=bool(elem.find('MAXROWS')),
+        # for some reason bool(elem.find('MAXROWS')) is incorrect
```
Comment (martinxsliu) [inline rets/http/parsers/parse.py]: We can remove this comment, or replace it with something like Python's `xml.etree.ElementTree.Element` objects are always considered false-y.

**Comment by martinxsliu** [comment]
Interesting, I was not aware of the RETS-STATUS tag. I think we should do an additional check for that when we current look for the status code in the root RETS tag.

File: rets/http/parsers/http.py [hunk 2]
```diff
@@ -25,14 +25,29 @@ def parse_response(response: Response_) -> Any:
...
+    if reply_code:
         raise RetsApiError(reply_code, reply_text, response.content)
 
     return root
 
 
+def parse_rets_status(elem: Element) -> Optional[Tuple[int, str]]:
+    rets_status = elem.find('RETS-STATUS')
+    if rets_status is None:
+        return None
```
Comment (martinxsliu) [inline rets/http/parsers/http.py]: Maybe we can have `rets_status = elem` here so we can avoid the duplication with the root.get('ReplyCode'). We should probably rename the `elem -> root` as well.

**Comment by martinxsliu** [comment]
Sorry, I acc

**Pull request merged**